bigbluebutton-html5/imports/ui/components/shortcut/component.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+
+export const withShortcut = (ComponentToWrap, shortcut) =>
+  class ShortcutWrapper extends Component {
+    constructor() {
+      super();
+
+      this.handleShortcut = this.handleShortcut.bind(this);
+    }
+
+    componentDidMount() {
+      document.addEventListener('keydown', this.handleShortcut, false);
+    }
+
+    componentWillUnmount() {
+      document.removeEventListener('keydown', this.handleShortcut, false);
+    }
+
+    handleShortcut(event) {
+      const combo = shortcut.toUpperCase();
+      const keys = combo.split('+');
+
+      if (event.key.toUpperCase() !== keys[keys.length - 1]) return;
+      if (!event.ctrlKey && (combo.includes('CONTROL') || combo.includes('CTRL'))) return;
+      if (!event.altKey && (combo.includes('ALT') || combo.includes('OPTION'))) return;
+      if (!event.shiftKey && keys.includes('SHIFT')) return;
+
+      this.element.ref.props.onClick();
```

Comment:
I still think this is a bad idea. You should fire a click event but without calling the prop `onClick`. What if the handler has another name? Or is being bounded on the `componentMount`?

I dont see why [`HTMLElement.click()`](HTMLElement.click()) would work in this case.

`this.element.click()`

Reply:
`this.element` is the ref of the `ComponentToWrap` but you are using a `ref` of the `element` ref? This HoC not only assumes that you have a `onClick` prop but that you have a `ref` prop too? That doesn't seems right.

Reply:
this.element.click() doesn't work in this case because this.element is not an HTMLElement